plugins/net_plugin/net_plugin.cpp

```diff
@@ -4636,6 +4542,14 @@ namespace eosio {
 
          cc.aggregated_vote().connect( broadcast_vote );
          cc.voted_block().connect( broadcast_vote );
+
+         if (bp_gossip_enabled()) {
+            cc.set_peer_keys_retrieval_active(true);
+            // update peer public keys from chainbase db
+            cc.update_peer_keys(cc.head().irreversible_blocknum());
+            // pass in empty set so validation that peer key exists is enforced
```

Comment:
that passing of empty set is not necessary or useful afaict.
Since this is the first time we call `cc.update_peer_keys()`, if the key can be found using `cc.get_peer_key(e)`, then it will also be present in the returned `flat_set`.

Reply:
If the node configures a `p2p-producer-peer` that is not registered on-chain then we want an error on startup.

Reply:
Ah yes, I just realized that thanks!